api/src/context.rs

```diff
@@ -405,17 +405,32 @@ impl Context {
     pub fn get_events(
         &self,
         event_key: &EventKey,
-        start: u64,
+        start: Option<u64>,
         limit: u16,
         ledger_version: u64,
     ) -> Result<Vec<EventWithVersion>> {
-        let events = self
-            .db
-            .get_events(event_key, start, Order::Ascending, limit as u64)?;
-        Ok(events
-            .into_iter()
-            .filter(|event| event.transaction_version <= ledger_version)
-            .collect::<Vec<_>>())
+        if let Some(start) = start {
```

Comment:
https://github.com/aptos-labs/aptos-core/pull/2665 

^ this'll make it clean, right?